osquery/config/config.cpp

```diff
@@ -137,17 +156,19 @@ std::map<std::string, std::vector<std::string> >& Config::getWatchedFiles() {
   return getInstance().cfg_.eventFiles;
 }
 
+pt::ptree& Config::getEntireConfiguration() {
+  boost::shared_lock<boost::shared_mutex> lock(rw_lock);
+  return getInstance().cfg_.all_data;
+}
+
 Status Config::getMD5(std::string& hash_string) {
-  std::string config_string;
-  auto s = genConfig(config_string);
-  if (!s.ok()) {
-    return s;
-  }
+  std::stringstream out;
+  write_json(out, getEntireConfiguration());
 
   hash_string = osquery::hashFromBuffer(
-      HASH_TYPE_MD5, (void*)config_string.c_str(), config_string.length());
+      HASH_TYPE_MD5, (void*)out.str().c_str(), out.str().length());
 
-  return Status(0, "OK");
+  return Status(1, "NI");
```

Comment:
What does this mean? On error return some meaningful, expect-to-be-printed-to-stderr-statement(tm).